### DISCUSSION TITLE
Edge function invocation from nextjs frontend (browser) not working

### DISCUSSION THREAD

**Discussion by CerealNotFound** [supabase/supabase#19148] [Questions]
I am trying to invoke an edge function deployed on supabase but it does not seem to work when called from the browser :/

It works fine when I call the API endpoint of the nextjs app from postman and I get the expected output, but it returns an empty array when I call it from the browser (expected output: array of objects with certain key-value pairs)

`export const corsHeaders = {
  "Access-Control-Allow-Origin": "*",
  "Access-Control-Allow-Headers":
    "authorization, x-client-info, apikey, content-type",
};

const { data, error } = await supabase.functions.invoke(
      "similarity_search",
      {
        body: JSON.stringify({
          query_embedding: query_embedding,
          pdf_id: "200511401",
        }),
        headers: { ...corsHeaders },
      }
    );`

The above is the relevant code for the invocation. 

Edge function logs on calling the request from the browser:
![image](https://github.com/supabase/supabase/assets/72287529/bda0fec1-dc0c-4c85-8a48-294a50301523)

**Comment by roychung96** [comment]
In the end I just do a helper function to call haiz
![image](https://github.com/supabase/supabase/assets/17947488/6c6b8158-ef00-4cd1-8153-ef4a1f9d09df)
